fails to build on ppc64le and s390x

Compilation fails on ppc64le and s390x (tested on Fedora):
```
$ /usr/bin/cmake -S . -B redhat-linux-build -DCMAKE_C_FLAGS_RELEASE:STRING=-DNDEBUG -DCMAKE_CXX_FLAGS_RELEASE:STRING=-DNDEBUG -DCMAKE_Fortran_FLAGS_RELEASE:STRING=-DNDEBUG -DCMAKE_VERBOSE_MAKEFILE:BOOL=ON -DCMAKE_INSTALL_DO_STRIP:BOOL=OFF -DCMAKE_INSTALL_PREFIX:PATH=/usr -DCMAKE_INSTALL_FULL_SBINDIR:PATH=/usr/bin -DCMAKE_INSTALL_SBINDIR:PATH=bin -DINCLUDE_INSTALL_DIR:PATH=/usr/include -DLIB_INSTALL_DIR:PATH=/usr/lib64 -DSYSCONF_INSTALL_DIR:PATH=/etc -DSHARE_INSTALL_PREFIX:PATH=/usr/share -DLIB_SUFFIX=64 -DBUILD_SHARED_LIBS:BOOL=ON -G Ninja -DCMAKE_BUILD_TYPE=RelWithDebInfo -DCMAKE_SKIP_INSTALL_RPATH=ON -DOAPV_APP_STATIC_BUILD=OFF
CMake Warning at CMakeLists.txt:28 (message):
  You have no git tool installed.
Version string has been taken from version.txt file.
OAPV VERSION=0.1.13
-- The C compiler identification is GNU 15.1.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
CMAKE_BUILD_TYPE=RelWithDebInfo
c Flags: -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m64 -march=z13 -mtune=z14 -fasynchronous-unwind-tables -fstack-clash-protection -DNDEBUG -O3 -fomit-frame-pointer -pthread -std=c99 -Wall -Wno-unused-function -Wno-pointer-sign -Wno-pointer-to-int-cast
linker Flags: -Wl,-z,relro -Wl,--as-needed   -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -specs=/usr/lib/rpm/redhat/redhat-hardened-ld-errors -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -Wl,--build-id=sha1 -specs=/usr/lib/rpm/redhat/redhat-package-notes -lm
SYSTEM_PROCESSOR=s390x
-- Performing Test COMPILER_HAS_HIDDEN_VISIBILITY
-- Performing Test COMPILER_HAS_HIDDEN_VISIBILITY - Success
-- Performing Test COMPILER_HAS_HIDDEN_INLINE_VISIBILITY
-- Performing Test COMPILER_HAS_HIDDEN_INLINE_VISIBILITY - Failed
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR - Success
-- Linux system
-- Red Hat-based system detected
-- Use RPM generator while generating installation package using CPack
-- Configuring done (0.5s)
-- Generating done (0.0s)
CMake Warning:
  Manually-specified variables were not used by the project:
    CMAKE_CXX_FLAGS_RELEASE
    CMAKE_C_FLAGS_RELEASE
    CMAKE_Fortran_FLAGS_RELEASE
    CMAKE_INSTALL_DO_STRIP
    CMAKE_VERBOSE_MAKEFILE
    INCLUDE_INSTALL_DIR
    LIB_INSTALL_DIR
    LIB_SUFFIX
    SHARE_INSTALL_PREFIX
    SYSCONF_INSTALL_DIR
-- Build files have been written to: /builddir/build/BUILD/openapv-0.1.13.1-build/openapv-0.1.13.1/redhat-linux-build
$ /usr/bin/cmake --build redhat-linux-build -j3 --verbose
...
[11/39] /usr/bin/gcc -Doapv_dynamic_EXPORTS -I/builddir/build/BUILD/openapv-0.1.13.1-build/openapv-0.1.13.1/redhat-linux-build/include -I/builddir/build/BUILD/openapv-0.1.13.1-build/openapv-0.1.13.1/src/oapv -I/builddir/build/BUILD/openapv-0.1.13.1-build/openapv-0.1.13.1/src/PUBLIC -I/builddir/build/BUILD/openapv-0.1.13.1-build/openapv-0.1.13.1/src/. -I/builddir/build/BUILD/openapv-0.1.13.1-build/openapv-0.1.13.1/src/.. -I/builddir/build/BUILD/openapv-0.1.13.1-build/openapv-0.1.13.1/src/../inc -I/builddir/build/BUILD/openapv-0.1.13.1-build/openapv-0.1.13.1/src/./sse -I/builddir/build/BUILD/openapv-0.1.13.1-build/openapv-0.1.13.1/src/./avx -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m64 -march=z13 -mtune=z14 -fasynchronous-unwind-tables -fstack-clash-protection -DNDEBUG -O3 -fomit-frame-pointer -pthread -std=c99 -Wall -Wno-unused-function -Wno-pointer-sign -Wno-pointer-to-int-cast -O2 -g -DNDEBUG -std=gnu99 -fPIC -msse4.1 -MD -MT src/CMakeFiles/oapv_dynamic.dir/sse/oapv_sad_sse.c.o -MF src/CMakeFiles/oapv_dynamic.dir/sse/oapv_sad_sse.c.o.d -o src/CMakeFiles/oapv_dynamic.dir/sse/oapv_sad_sse.c.o -c /builddir/build/BUILD/openapv-0.1.13.1-build/openapv-0.1.13.1/src/sse/oapv_sad_sse.c
FAILED: src/CMakeFiles/oapv_dynamic.dir/sse/oapv_sad_sse.c.o 
/usr/bin/gcc -Doapv_dynamic_EXPORTS -I/builddir/build/BUILD/openapv-0.1.13.1-build/openapv-0.1.13.1/redhat-linux-build/include -I/builddir/build/BUILD/openapv-0.1.13.1-build/openapv-0.1.13.1/src/oapv -I/builddir/build/BUILD/openapv-0.1.13.1-build/openapv-0.1.13.1/src/PUBLIC -I/builddir/build/BUILD/openapv-0.1.13.1-build/openapv-0.1.13.1/src/. -I/builddir/build/BUILD/openapv-0.1.13.1-build/openapv-0.1.13.1/src/.. -I/builddir/build/BUILD/openapv-0.1.13.1-build/openapv-0.1.13.1/src/../inc -I/builddir/build/BUILD/openapv-0.1.13.1-build/openapv-0.1.13.1/src/./sse -I/builddir/build/BUILD/openapv-0.1.13.1-build/openapv-0.1.13.1/src/./avx -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m64 -march=z13 -mtune=z14 -fasynchronous-unwind-tables -fstack-clash-protection -DNDEBUG -O3 -fomit-frame-pointer -pthread -std=c99 -Wall -Wno-unused-function -Wno-pointer-sign -Wno-pointer-to-int-cast -O2 -g -DNDEBUG -std=gnu99 -fPIC -msse4.1 -MD -MT src/CMakeFiles/oapv_dynamic.dir/sse/oapv_sad_sse.c.o -MF src/CMakeFiles/oapv_dynamic.dir/sse/oapv_sad_sse.c.o.d -o src/CMakeFiles/oapv_dynamic.dir/sse/oapv_sad_sse.c.o -c /builddir/build/BUILD/openapv-0.1.13.1-build/openapv-0.1.13.1/src/sse/oapv_sad_sse.c
cc1: error: unrecognized command-line option ‘-msse4.1’
ninja: build stopped: subcommand failed.
```

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -39,11 +39,14 @@ if(ARM)
                                       ${LIB_NEON_SRC} ${LIB_NEON_INC} )
   add_library( ${LIB_NAME_BASE}_dynamic SHARED ${LIB_API_SRC} ${LIB_INC} ${LIB_BASE_SRC} ${LIB_BASE_INC}
                                               ${LIB_NEON_SRC} ${LIB_NEON_INC} )
-else()
+elseif(X86)
   add_library( ${LIB_NAME_BASE} STATIC ${LIB_API_SRC} ${LIB_INC} ${LIB_BASE_SRC} ${LIB_BASE_INC} ${LIB_SSE_SRC} ${LIB_SSE_INC}
                                       ${LIB_AVX_SRC} ${LIB_AVX_INC} )
   add_library( ${LIB_NAME_BASE}_dynamic SHARED ${LIB_API_SRC} ${LIB_INC} ${LIB_BASE_SRC} ${LIB_BASE_INC} ${LIB_SSE_SRC} ${LIB_SSE_INC}
                                               ${LIB_AVX_SRC} ${LIB_AVX_INC} )
+else()
+  add_library( ${LIB_NAME_BASE} STATIC ${LIB_API_SRC} ${LIB_INC} ${LIB_BASE_SRC} ${LIB_BASE_INC} )
+  add_library( ${LIB_NAME_BASE}_dynamic SHARED ${LIB_API_SRC} ${LIB_INC} ${LIB_BASE_SRC} ${LIB_BASE_INC} )
 endif()
 
 set_target_properties(${LIB_NAME_BASE}_dynamic PROPERTIES VERSION ${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR} SOVERSION ${LIB_SOVERSION})
@@ -78,8 +81,10 @@ source_group("base\\avx\\source" FILES ${LIB_AVX_SRC})
 
 if(ARM)
   include_directories( ${LIB_NAME_BASE} PUBLIC . .. ../inc ./neon)
-else()
+elseif(X86)
   include_directories( ${LIB_NAME_BASE} PUBLIC . .. ../inc ./sse ./avx)
+else()
+  include_directories( ${LIB_NAME_BASE} PUBLIC . .. ../inc)
 endif()
 
 set_target_properties(${LIB_NAME_BASE} PROPERTIES FOLDER lib
@@ -98,7 +103,7 @@ elseif( UNIX OR MINGW )
 
   if (ARM)
     set_property( SOURCE ${NEON} APPEND PROPERTY COMPILE_FLAGS "-flax-vector-conversions" )
-  else()
+  elseif (X86)
     set_property( SOURCE ${SSE} APPEND PROPERTY COMPILE_FLAGS "-msse4.1" )
     set_property( SOURCE ${AVX} APPEND PROPERTY COMPILE_FLAGS " -mavx2" )
   endif()

